Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,324 @@
+                }
+                return []
+            }
+            let ranks = values.map { shape in (shape?.rank).map(Int32.init) ?? -1 }
+            flattenedDims.withUnsafeBufferPointer { flattenedDimsBuffer in
+                var dimsPtr: UnsafePointer<Int64>? = flattenedDimsBuffer.baseAddress
+                var dims: [UnsafePointer<Int64>?] = []
+                for rank in ranks {
+                    dims.append(dimsPtr)
+                    if rank >= 0 {
+                        dimsPtr = dimsPtr.map { $0.advanced(by: Int(rank)) }
+                    }
+                }
+                dims.withUnsafeMutableBufferPointer { dimsBuffer in
+                    ranks.withUnsafeBufferPointer { ranksBuffer in
+                        TF_SetAttrShapeList(
+                            desc,
+                            name,
+                            dimsBuffer.baseAddress,
+                            ranksBuffer.baseAddress,
+                            Int32(ranksBuffer.count))
+                    }
+                }
+            }
+        default: assert(false, "Unhandled attribute \(name):\(attrValue)")
+        }
+    }
+
+    private func makeTFGraphNode(
+        name: String,
+        attrs: [String: LazyTensorOperation.Attribute],
+        inputs: [Input],
+        device: String?
+    ) -> CTFOperation? {
+        // Create a new graph node now.
+        let desc: CTFOperationDescription! = TF_NewOperation(
+            cTFGraph,
+            name,
+            newNodeName(base: name))
+
+        // Set Attributes
+        for (name, value) in attrs {
+            updateAttribute(desc, name, value)
+        }
+
+        // Add Inputs
+        for input in inputs {
+            switch input {
+            case Input.single(let singleInput):
+                TF_AddInput(desc, singleInput)
+            case Input.list(let inputList):
+                inputList.withUnsafeBufferPointer { buffer in
+                    TF_AddInputList(desc, buffer.baseAddress, Int32(buffer.count))
+                }
+            }
+        }
+
+        if let device = device { TF_SetDevice(desc, device) }
+
+        // Finalize operation.
+        let graphNode = TF_FinishOperation(desc, status)
+        checkOk(status)
+        return graphNode!
+    }
+
+    private func makeTFConstNode(_ handle: TFETensorHandle) -> TF_Output {
+        let cTensorHandle = handle._cTensorHandle
+        let cTensor = TFE_TensorHandleResolve(cTensorHandle, status)
+        checkOk(status)
+        let desc = TF_NewOperation(cTFGraph, "Const", newNodeName(base: "Const"))
+        checkOk(status)
+        TF_SetAttrType(desc, "dtype", TFE_TensorHandleDataType(cTensorHandle))
+        TF_SetAttrTensor(desc, "value", cTensor, status)
+        checkOk(status)
+        let constNode = TF_FinishOperation(desc, status)
+        return TF_Output(oper: constNode, index: 0)
+    }
+
+    private func makeTFOutput(
+        _ lazyHandle: LazyTensor,
```

Comment:
Use argument labels in factory methods (methods whose names start with "make").

```suggestion
        handle: LazyTensor,
```